Wine/Mingw unit tests crash

The Wine/Mingw unit tests running on CircleCI randomly (and usually) crash. This currently affects master. Sample output is:
```
#!/bin/bash -eo pipefail
make --keep-going check
wine .build/testBin/runTests
wine: Unhandled page fault on write access to 0x00000000 at address 0x4b3a44 (thread 0009), starting debugger...
002e:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
002e:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
Unhandled exception: page fault on write access to 0x00000000 in 32-bit code (0x004b3a44).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:006b GS:0063
 EIP:004b3a44 ESP:00b5fc28 EBP:00b5fc28 EFLAGS:00010246(  R- --  I  Z- -P- )
 EAX:00000000 EBX:00b5fd78 ECX:0054f060 EDX:00000000
 ESI:00241318 EDI:00000028
Stack dump:
0x00b5fc28:  00b5fc68 005022bc 00000000 00b5fc5f
0x00b5fc38:  00b5fd48 004375f4 005335d0 0053431e
0x00b5fc48:  00b5fc68 0054f060 00b5fd80 00000000
0x00b5fc58:  00b5fd80 0054f060 00241318 00b5fd78
0x00b5fc68:  00b5fca8 00504300 00000000 7b4638f6
0x00b5fc78:  00b5fca0 00000000 00241318 7b4638f6
Backtrace:
=>0 0x004b3a44 in runtests (+0xb3a44) (0x00b5fc28)
  1 0x005022bc in runtests (+0x1022bb) (0x00b5fc68)
  2 0x00504300 in runtests (+0x1042ff) (0x00b5fca8)
  3 0x00502fc9 in runtests (+0x102fc8) (0x00b5fcd8)
  4 0x005055d4 in runtests (+0x1055d3) (0x00b5fd48)
  5 0x004079a1 in runtests (+0x79a0) (0x00b5fd98)
  6 0x0040a3c9 in runtests (+0xa3c8) (0x00b5fdb8)
  7 0x0040a3fc in runtests (+0xa3fb) (0x00b5fdd8)
  8 0x00435afa in runtests (+0x35af9) (0x00b5fec0)
  9 0x004013b7 in runtests (+0x13b6) (0x00b5fec0)
  10 0x7b461a7c in kernel32 (+0x51a7b) (0x00b5fed8)
  11 0x7b4634ce in kernel32 (+0x534cd) (0x00b5ffd8)
  12 0x7b461a8a in kernel32 (+0x51a89) (0x00b5ffec)
0x004b3a44: movb	%dl,0x0(%eax)
Modules:
Module	Address			Debug info	Name (11 modules)
PE	  400000-  951000	Dwarf           runtests
PE	7b410000-7b5b6000	Export          kernel32
PE	7bc10000-7bc14000	Deferred        ntdll
PE	7f5d0000-7f5d4000	Deferred        imm32
PE	7f780000-7f784000	Deferred        version
PE	7f7a0000-7f7a4000	Deferred        advapi32
PE	7f820000-7f827000	Deferred        gdi32
PE	7f950000-7fa16000	Deferred        user32
PE	7fb30000-7fb38000	Deferred        shlwapi
PE	7fba0000-7fd02000	Deferred        shell32
PE	7fe00000-7fe04000	Deferred        msvcrt
Threads:
process  tid      prio (all id:s are in hex)
00000008 (D) Z:\root\project\.build\testBin\runTests
	00000009    0 <==
0000000c winemenubuilder.exe
	0000000d    0
0000000e services.exe
	00000027    0
	00000020    0
	0000001a    0
	00000017    0
	00000016    0
	00000015    0
	00000010    0
	0000000f    0
00000011 explorer.exe
	00000026    0
	00000025    0
	00000012    0
00000013 winedevice.exe
	0000001c    0
	0000001b    0
	00000019    0
	00000018    0
	00000014    0
0000001d plugplay.exe
	00000022    0
	00000021    0
	0000001e    0
00000023 winedevice.exe
	0000002c    0
	0000002b    0
	0000002a    0
	00000029    0
	00000028    0
	00000024    0
System information:
    Wine build: wine-3.0 (Ubuntu 3.0-1ubuntu1)
    Platform: i386
    Version: Windows 7
    Host system: Linux
    Host version: 4.15.0-1035-aws
makefile:121: recipe for target 'check' failed
make: *** [check] Error 5
Exited with code 2
```

Ideally the root cause should be determined and the problem fixed. Failing that, or if diagnosing or fixing the problem takes too long, we should consider disabling the unit test runs on CircleCI. We could still build both the main project and the test project, but would stop before running the unit tests. Alternatively, we could try re-running the unit tests if they appear to have failed due to a crash.

----

Possible causes (mostly just wild guesses):
Lack of graphical environment on CircleCI, which Wine might implicitly use
Inappropriate initialization or destruction of a global variable not suitable for the test code
Interference from patch code triggering in unit test executable
Bad test code
Bad compile or improper linking
Bug in Wine

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -9,4 +9,4 @@ jobs:
       - run: git submodule update --init || true
       - run: make --keep-going
       - run: make --keep-going test
-#      - run: make --keep-going check
+      - run: make --keep-going check

--- a/srcStatic/ConsoleModuleLoader.cpp
+++ b/srcStatic/ConsoleModuleLoader.cpp
@@ -11,7 +11,9 @@
 #include <cstddef>
 #include <system_error>
 
-std::string moduleDirectory;
+
+extern std::string moduleDirectory;
+
 
 ConsoleModuleLoader::ConsoleModuleLoader(const std::string& moduleRelativeDirectory)
 {

--- a/srcStatic/op2ext-Internal.cpp
+++ b/srcStatic/op2ext-Internal.cpp
@@ -6,14 +6,15 @@
 
 
 bool modulesRunning = false;
+Logger logger;
+VolList volList;
+std::string moduleDirectory; // Must be defined + initialized before consoleModLoader
+ConsoleModuleLoader consoleModLoader(FindModuleDirectory());
+IniModuleLoader iniModuleLoader;
+
 
 void EnableTestEnvironment()
 {
 	DisableModalDialogs();
 	DisableMemoryCommands();
 }
-
-Logger logger;
-VolList volList;
-ConsoleModuleLoader consoleModLoader(FindModuleDirectory());
-IniModuleLoader iniModuleLoader;

